agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/LoggerExporterCustomizer.java

```diff
@@ -0,0 +1,98 @@
+ */
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import com.google.auto.service.AutoService;
+import com.microsoft.applicationinsights.agent.internal.configuration.Configuration;
+import com.microsoft.applicationinsights.agent.internal.exporter.AiOperationNameLogWrappingProcessor;
+import com.microsoft.applicationinsights.agent.internal.exporter.LoggerExporter;
+import com.microsoft.applicationinsights.agent.internal.processors.ExporterWithLogProcessor;
+import com.microsoft.applicationinsights.agent.internal.processors.LogExporterWithAttributeProcessor;
+import com.microsoft.applicationinsights.agent.internal.telemetry.TelemetryClient;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.logs.LogProcessor;
+import io.opentelemetry.sdk.logs.export.BatchLogProcessor;
+import io.opentelemetry.sdk.logs.export.LogExporter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AutoService(AutoConfigurationCustomizerProvider.class)
+public class LoggerExporterCustomizer implements AutoConfigurationCustomizerProvider {
+
+  @Override
+  public void customize(AutoConfigurationCustomizer autoConfiguration) {
+    TelemetryClient telemetryClient = TelemetryClient.getActive();
+    if (telemetryClient == null) {
+      // agent failed during startup
+      return;
+    }
+
+    autoConfiguration.addLogEmitterProviderCustomizer(
+        (builder, config) -> {
+          List<Configuration.ProcessorConfig> processorConfigs =
+              reverseProcessorConfigs(MainEntryPoint.getConfiguration());
+          return builder.addLogProcessor(createLogExporter(processorConfigs));
+        });
+  }
+
+  private static LogProcessor createLogExporter(
+      List<Configuration.ProcessorConfig> processorConfigs) {
+    LogExporter logExporter = new LoggerExporter(TelemetryClient.getActive());
+    if (!processorConfigs.isEmpty()) {
+      for (Configuration.ProcessorConfig processorConfig : processorConfigs) {
+        switch (processorConfig.type) {
+          case ATTRIBUTE:
+            logExporter = new LogExporterWithAttributeProcessor(processorConfig, logExporter);
+            break;
+          case LOG:
+            logExporter = new ExporterWithLogProcessor(processorConfig, logExporter);
+            break;
+          default:
+            throw new IllegalStateException(
+                "Not an expected ProcessorType: " + processorConfig.type);
+        }
+      }
+    }
+
+    // operation name need to be captured while on the main thread
+    // before passing off to the BatchLogProcessor
+    return new AiOperationNameLogWrappingProcessor(
+        BatchLogProcessor.builder(logExporter).setMaxExportBatchSize(1).build());
+  }
+
+  private static List<Configuration.ProcessorConfig> reverseProcessorConfigs(
```

Comment:
and I think the purpose of reversing is more clearer if moved inside of `createLogExporter`

```suggestion
  private static List<Configuration.ProcessorConfig> getLogProcessorConfigs(
```